### PR TITLE
chore[devtools]: upgrade to webpack v5

### PR DESCRIPTION
**Pull request by hoxyq**
## Summary
- Updated `webpack` (and all related packages) to v5 in `react-devtools-*` packages.
- I haven't touched any `TODO (Webpack 5)`. Tried to poke it, but each my attempt failed and parsing hook names feature stopped working. I will work on this in a separate PR.
- This work is one of prerequisites for updating Firefox extension to manifests v3

related PRs:
https://github.com/facebook/react/pull/22267
https://github.com/facebook/react/pull/26506

## How did you test this change?
Tested on all surfaces, explicitly checked that parsing hook names feature still works.

### CODE REVIEW

File: packages/react-devtools-extensions/build.js [hunk 1]
```diff
@@ -56,14 +56,7 @@ const build = async (tempPath, manifestPath) => {
...
 
-  const webpackPath = join(
-    __dirname,
-    '..',
-    '..',
-    'node_modules',
-    '.bin',
-    'webpack',
-  );
+  const webpackPath = join(__dirname, 'node_modules', '.bin', 'webpack');
```
Comment (hoxyq) [inline packages/react-devtools-extensions/build.js:59]: Not sure why we were using `webpack` from root `node_modules`.

Updated to use it from `react-devtools-extensions` package, where `webpack`'s version was updated. The root one is still using version 4.x.x.

File: packages/react-devtools-shell/webpack-server.js [hunk 2]
```diff
@@ -200,4 +182,48 @@ const e2eRegressionApp = semver.lt(REACT_VERSION, '18.0.0')
...
+  },
+  e2eRegressionAppCompiler,
+);
+
+const runServer = async () => {
+  console.log('Starting server...');
+
+  appServer.compiler.hooks.done.tap('done', () =>
+    console.log(SUCCESSFUL_COMPILATION_MESSAGE),
+  );
```
Comment (hoxyq) [inline packages/react-devtools-shell/webpack-server.js:223]: This is the easiest way that I found on how to emit something (like an event) when webpack has compiled assets.

More scalable solution for it would be wrapping up in a Promise, but this server is being started via spawning node process in `scripts/circleci/run_devtools_e2e_tests.js`, so we need something like this ¯\\\_(ツ)\_/¯

File: packages/react-devtools-shell/webpack-server.js [hunk 3]
```diff
@@ -25,12 +28,6 @@ if (!NODE_ENV) {
   process.exit(1);
 }
 
-const TARGET = process.env.TARGET;
```
Comment (tyao1) [inline packages/react-devtools-shell/webpack-server.js:28]: Why is this removed?

**Comment by hoxyq** [comment]
> > This work is one of prerequisites for updating Firefox extension to manifests v3
> 
> Just curious why it is for Firefox, and is it a prerequisite for Chrome extension?

> Why it is for Firefox?

When I was trying to upgrade Firefox extension to manifest v3, I was getting some errors related to `@babel/preset-env` package not supporting some api's for `firefox` target, don't remember the exact problem. In order to fix it, we need to update `@babel/preset-env`, but next supported version with a fix requires updated versions of `babel-loader` package, which works only on webpack v5 - https://webpack.js.org/loaders/babel-loader/#install.

There is a decent chance that this error was not related to our old webpack version, but upgrading it allows us to upgrade other babel packages now.

This is not the only blocker for migrating to manifest v3 on Firefox, the main one is that Firefox doesn't support [ExecutionWorld.MAIN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/ExecutionWorld) yet - https://discourse.mozilla.org/t/will-scripting-executionworld-main-be-supported/112919

> Is it a prerequisite for Chrome extension?

No, Chrome extension already migrated to manifest v3.

File: packages/react-devtools-shell/webpack-server.js (hunk 3 above)
Comment (hoxyq) [reply packages/react-devtools-shell/webpack-server.js:28]: Because its not needed, we are not building `react-devtools-shell` to static files, it is only used as a playground to test inline version of DevTools.

So the only target is `local` (to start dev-server)

**Comment by hoxyq** [comment]
There is a regression in parsing hooks names for React v17+, don't know why CI tests didn't catch that. Looking for ways to resolve it.

**Comment by hoxyq** [comment]
> There is a regression in parsing hooks names for React v17+, don't know why CI tests didn't catch that. Looking for ways to resolve it.

Now fixed, there was a problem with CORS policy in testing shell, which we start with webpack dev server.
